neo/public/javascripts/map/controllers.js

```diff
@@ -336,34 +374,99 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
         });
     }
 
-    function loadCityJsonByBound(onEachFeature){
-      var bounds = $scope.map.getBounds();
-      var rteBounds = "city/" + bounds._northEast.lat + "/" + bounds._southWest.lat + "/" + bounds._northEast.lng + "/" + bounds._southWest.lng;
-      $http.get(rteBounds)
-        .success(function(data) {
-          $scope.geojsonData.city = data;
-          if($scope.polygons.cityPolygons) {
-            $scope.map.removeLayer($scope.polygons.cityPolygons);
-          }
-          $scope.polygons.cityPolygons = L.geoJson(data, {
-            style: $scope.styles.cityStyle,
-            onEachFeature: onEachFeature
-          });
-          setCenterAndBoundry($scope.geojsonData.city.features);
-          if (!$scope.status.init) {
-            resetGeoIds($scope.bounds, $scope.geojsonData.city, 'cityID');
-            Asterix.parameters.geoLevel = 'city';
-            Asterix.queryType = 'zoom';
-            Asterix.query(Asterix.parameters, Asterix.queryType);
-          }
-          $scope.map.addLayer($scope.polygons.cityPolygons);
-        })
-        .error(function(data) {
-          console.error("Load city data failure");
-        });
-    }
 
 
+
+        function loadCityJsonByBound(onEachFeature){
+        
+        var bounds = $scope.map.getBounds();
+        var data_response;
+
+        var rteBounds = "city/" + bounds._northEast.lat + "/" + bounds._southWest.lat + "/" + bounds._northEast.lng + "/" + bounds._southWest.lng;
+
+         var poly1 = {
+                                 "type": "Feature",
+
+                                 "geometry": {
+                                   "type": "Polygon",
+                                   "coordinates": [[
+                                      [bounds._northEast.lng,bounds._northEast.lat],
+                                      [bounds._northEast.lng,bounds._southWest.lat],
+                                      [bounds._southWest.lng,bounds._southWest.lat],
+                                      [bounds._southWest.lng,bounds._northEast.lat],
+                                      [bounds._northEast.lng,bounds._northEast.lat]
+                                   ]]
+                                 },
+                                 "properties":{
+                                            "centerLog" : 0,
+                                             "centerLat" : 0
+
+                                 }
+                      };
+        var pt1 = turf.point([ bounds._northEast.lng ,bounds._northEast.lat]);
+        var pt2 = turf.point([bounds._southWest.lng,bounds._southWest.lat ]);
+        var bbox = turf.bbox(poly1);
+        
+        var item = {
+             minX : bbox[0],
+             minY : bbox[1],
+             maxX : bbox[2],
+             maxY : bbox[3]
+                  };
+
+
+        if($scope.cachecount>1 && turf.inside(pt1,$scope.cache) && turf.inside(pt2,$scope.cache))
```

Comment:
for this kind of complex condition match, it's better to make a separate meaningful function like `isPolygonRequestCached` and call that function to be clear.